src/commands/database/list.mjs

```diff
@@ -1,33 +1,98 @@
 
-async function listDatabases() {
-  const logger = container.resolve("logger");
+// Narrow the output fields based on the provided flags.
+const getOutputFields = (argv) => {
+  if (!argv.secret && !argv.database) {
+    // If we are listing top level databases the region group
+    // needs to be included as database names can be re-used across
+    // regions.
+    return ["name", "region_group"];
+  }
+  return ["name"];
+};
 
-  // query the account api
-  const accountClient = new FaunaAccountClient();
-  const databases = await accountClient.listDatabases();
-  logger.stdout(databases);
-
-  // see what credentials are being used
-  const credentials = container.resolve("credentials");
-  logger.debug(
-    `
-    Account Key: ${credentials.accountKeys.key}\n
-    Database Key: ${credentials.databaseKeys.key}`,
-    "creds",
+function pickOutputFields(databases, argv) {
+  return databases.map((d) =>
+    getOutputFields(argv).reduce((acc, field) => {
+      acc[field] = d[field];
+      return acc;
+    }, {}),
   );
 }
 
+async function listDatabasesWithAccountAPI(argv) {
+  const { pageSize, database, json } = argv;
+  const accountClient = new FaunaAccountClient();
+  const response = await accountClient.listDatabases({
+    pageSize,
+    path: database,
+  });
+  const output = pickOutputFields(response.results, argv);
+  if (json) {
+    container.resolve("logger").stdout(JSON.stringify(output));
+  } else {
+    container.resolve("logger").stdout(output);
+  }
+}
+
+async function listDatabasesWithSecret(argv) {
+  const { url, secret, pageSize, json } = argv;
+  const { runQueryFromString, formatQueryResponse } =
+    container.resolve("faunaClientV10");
+
+  try {
+    const result = await runQueryFromString({
+      url,
+      secret,
+      // This gives us back an array of database names. If we want to
+      // provide the after token at some point this query will need to be updated.
+      expression: `Database.all().paginate(${pageSize}).data { ${getOutputFields(argv)} }`,
+    });
+    container.resolve("logger").stdout(formatQueryResponse(result, { json }));
+  } catch (e) {
+    if (e instanceof FaunaError) {
+      throwForError(e);
+    }
+    throw e;
+  }
+}
+
+async function listDatabases(argv) {
+  if (argv.secret) {
+    return listDatabasesWithSecret(argv);
+  } else {
+    return listDatabasesWithAccountAPI(argv);
+  }
+}
+
 function buildListCommand(yargs) {
   return yargs
     .options({
-      ...commonQueryOptions,
+      pageSize: {
+        type: "number",
+        default: 1000,
+      },
     })
-    .help("help", "show help");
+    .help("help", "show help")
+    .example([
+      ["$0 database list", "list all databases"],
+      [
+        "$0 database list --database 'us-std/example'",
+        "list all databases under us-std/example",
+      ],
+      [
+        "$0 database list --secret 'my-secret'",
+        "list all databases using the provided database secret",
```

Comment:
what does `fauna database list --secret 'my-secret' --database 'eu-std/example'` do?

Reply:
You get this error message `Cannot use both the '--secret' and '--database' options together. Please specify only one.`